integrations/venice-samza/src/main/java/com/linkedin/venice/samza/VeniceSystemProducer.java

```diff
@@ -136,7 +137,8 @@ public class VeniceSystemProducer implements SystemProducer, Closeable {
   private Schema keySchema;
   private String canonicalKeySchemaStr;
   // To avoid the excessive usage of the cache in case each message is using a unique key schema
-  private final Map<Schema, String> canonicalSchemaStrCache = new BoundedHashMap<>(10, true);
+  private final LoadingCache<Schema, String> canonicalSchemaStrCache =
+      Caffeine.newBuilder().maximumSize(10).build(AvroCompatibilityHelper::toParsingForm);
```

Comment:
Can we bump this to a bit larger number ? like 50.. 10 seems too conservative.

Reply:
It should be more enough for most of the use cases and this is mainly to put some safeguard for some abusing use cases.